src/components/Reactions/ReportActionItemEmojiReactions.js

```diff
@@ -132,24 +138,29 @@ function ReportActionItemEmojiReactions(props) {
                             key={reaction.reactionEmojiName}
                         >
                             <View>
-                                <EmojiReactionBubble
-                                    ref={(ref) => (popoverReactionListAnchors.current[reaction.reactionEmojiName] = ref)}
-                                    count={reaction.reactionCount}
-                                    emojiCodes={reaction.emojiCodes}
-                                    onPress={reaction.onPress}
-                                    reactionUsers={reaction.reactionUsers}
-                                    hasUserReacted={reaction.hasUserReacted}
-                                    onReactionListOpen={reaction.onReactionListOpen}
-                                    shouldBlockReactions={props.shouldBlockReactions}
-                                />
+                                <OfflineWithFeedback
+                                    pendingAction={reaction.hasPendingAction}
+                                    shouldDisableOpacity={Boolean(reportAction.pendingAction)}
+                                >
+                                    <EmojiReactionBubble
+                                        ref={(ref) => (popoverReactionListAnchors.current[reaction.reactionEmojiName] = ref)}
+                                        count={reaction.reactionCount}
+                                        emojiCodes={reaction.emojiCodes}
+                                        onPress={reaction.onPress}
+                                        reactionUsers={reaction.reactionUsers}
+                                        hasUserReacted={reaction.hasUserReacted}
+                                        onReactionListOpen={reaction.onReactionListOpen}
+                                        shouldBlockReactions={props.shouldBlockReactions}
+                                    />
+                                </OfflineWithFeedback>
                             </View>
                         </Tooltip>
                     );
                 })}
                 {!props.shouldBlockReactions && (
                     <AddReactionBubble
                         onSelectEmoji={props.toggleReaction}
-                        reportAction={{reportActionID: props.reportActionID}}
+                        reportAction={reportAction}
```

Comment:
Just glancing over this, I think `AddReactionBubble` will need to be updated too?

Reply:
Can you double check that you've updated all references to the `reportActionID`?

Reply:
Yep I did check, the `AddReactionBubble` was already expecting a kind of partial object similar to reportAction 

```
    reportAction: PropTypes.shape({
        reportActionID: PropTypes.string.isRequired,
    }),
```
The contract is already fullfilled by `reportAction` object that we are passing now.

Reply:
Yep I did check, the `AddReactionBubble` was already expecting a kind of partial object similar to reportAction 

```
    reportAction: PropTypes.shape({
        reportActionID: PropTypes.string.isRequired,
    }),
```
The contract is already fullfilled by `reportAction` object that we are passing now.